crates/primitives/receipt/src/from_blockifier.rs

```diff
@@ -171,102 +184,3 @@ impl From<GasVector> for L1Gas {
-    };
-
-    loop {
-        if event_idx < call_info.execution.events.len() {
-            let ordered_event = &call_info.execution.events[event_idx];
-            if ordered_event.order == next_order {
-                let event = Event {
-                    from_address: call_info.call.storage_address.into(),
-                    keys: ordered_event.event.keys.iter().map(|k| k.0).collect(),
-                    data: ordered_event.event.data.0.clone(),
-                };
-                events_vec.push(event);
-                next_order += 1;
-                event_idx += 1;
-                continue;
-            }
-        }
-
-        if inner_call_idx < call_info.inner_calls.len() {
-            next_order =
-                get_events_from_call_info(Some(&call_info.inner_calls[inner_call_idx]), next_order, events_vec);
-            inner_call_idx += 1;
-            continue;
-        }
-
-        break;
-    }
-
-    next_order
-}
-
-#[cfg(test)]
-mod events_logic_tests {
-    use crate::from_blockifier::get_events_from_call_info;
-    use crate::Event;
-    use blockifier::execution::call_info::{CallExecution, CallInfo, OrderedEvent};
-    use rstest::rstest;
-    use starknet_api::transaction::{EventContent, EventData, EventKey};
-    use starknet_types_core::felt::Felt;
-
-    #[rstest]
-    fn test_event_ordering() {
-        let mut events = Vec::new();
-        let nested_calls = create_call_info(
-            0,
-            vec![create_call_info(
-                1,
-                vec![create_call_info(2, vec![create_call_info(3, vec![create_call_info(4, vec![])])])],
-            )],
-        );
-        get_events_from_call_info(Some(&nested_calls), 0, &mut events);
-
-        let expected_events_ordering = vec![event(0), event(1), event(2), event(3), event(4)];
-
-        assert_eq!(expected_events_ordering, events);
-    }
-
-    fn create_call_info(event_number: u32, inner_calls: Vec<CallInfo>) -> CallInfo {
-        CallInfo {
-            call: Default::default(),
-            execution: execution(vec![ordered_event(event_number as usize)]),
-            resources: Default::default(),
-            inner_calls,
-            storage_read_values: vec![],
-            accessed_storage_keys: Default::default(),
-        }
-    }
-
-    fn execution(events: Vec<OrderedEvent>) -> CallExecution {
-        CallExecution {
-            retdata: Default::default(),
-            events,
-            l2_to_l1_messages: vec![],
-            failed: false,
-            gas_consumed: Default::default(),
-        }
-    }
-
-    fn ordered_event(order: usize) -> OrderedEvent {
-        OrderedEvent {
-            order,
-            event: EventContent { keys: vec![EventKey(Felt::ZERO); order], data: EventData(vec![Felt::ZERO; order]) },
-        }
-    }
-
-    fn event(order: usize) -> Event {
-        Event { from_address: Default::default(), keys: vec![Felt::ZERO; order], data: vec![Felt::ZERO; order] }
-    }
-}
```

Comment:
Why remove this test? Now we do not check if `recursive_call_info_iter` is actually operating recursively.

Reply:
okay i'll leave it